### PR TITLE
refactor(filterable-select): convert unit tests to RTL

### PR DESCRIPTION
**Pull request by edleeks87**
### Proposed behaviour

Converts FilterableSelect unit tests to RTL

### Current behaviour

FilterableSelect unit tests are enzyme
 
### Checklist

- [x] Commits follow our style guide
- [x] Unit tests added or updated if required
- [x] Playwright automation tests added or updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

**Review by damienrobson-sage — APPROVED** [review]

File: src/components/select/filterable-select/filterable-select.test.tsx [hunk 1]
```diff
@@ -83,33 +220,1577 @@ test("combobox has correct accessible name when aria-labelledby prop is provided
...
+
+    expect(await screen.findByTestId("select-list-wrapper")).toHaveAttribute(
+      "data-floating-placement",
+      "bottom-end"
+    );
+  });
+});
+
+describe("when the user types in the input", () => {
+  it("should update the input value when the user presses 'Delete' and the text is hightlighted", async () => {
```
Comment (nuria1110) [inline src/components/select/filterable-select/filterable-select.test.tsx:1163]: typo
```suggestion
  it("should update the input value when the user presses 'Delete' and the text is highlighted", async () => {
```

File: src/components/select/filterable-select/filterable-select.test.tsx [hunk 2]
```diff
@@ -83,33 +220,1577 @@ test("combobox has correct accessible name when aria-labelledby prop is provided
...
+        onFilterChange={onFilterChangeFn}
+      >
+        <Option value="opt1" text="red" />
+      </FilterableSelect>
+    );
+
+    expect(onFilterChangeFn).not.toHaveBeenCalled();
+  });
+
+  it("foo", async () => {
```
Comment (nuria1110) [inline src/components/select/filterable-select/filterable-select.test.tsx:1343]: **question:** was this left by accident?

File: src/components/select/filterable-select/filterable-select.test.tsx [hunk 3]
```diff
@@ -83,33 +220,1577 @@ test("combobox has correct accessible name when aria-labelledby prop is provided
...
+  render(
+    <FilterableSelectWithState>
+      <Option value="opt1" text="a" />
+      <Option value="opt2" text="b" />
+      <Option value="opt3" text="c" />
+    </FilterableSelectWithState>
+  );
+  await user.type(screen.getByRole("combobox"), "a");
+
+  expect(await screen.findAllByRole("option")).toHaveLength(1);
```
Comment (nuria1110) [inline src/components/select/filterable-select/filterable-select.test.tsx:1795]: **nitpick (non-blocking):** I see the test descriptions are different but this test looks identical to the previous one, could add `disableDefaultFiltering={false}` the the previous test to match the description a bit more accurately.

File: src/components/select/filterable-select/filterable-select.test.tsx (hunk 2 above)
Comment (edleeks87) [reply src/components/select/filterable-select/filterable-select.test.tsx:1343]: yep 🤦

File: src/components/select/filterable-select/filterable-select.test.tsx (hunk 3 above)
Comment (edleeks87) [reply src/components/select/filterable-select/filterable-select.test.tsx:1795]: yep the one above should be explicitly be setting the prop I'll update thanks

**Review by damienrobson-sage — DISMISSED** [review]

**Review by nuria1110 — APPROVED** [review]

File: src/components/select/filterable-select/filterable-select.test.tsx [hunk 4]
```diff
@@ -83,33 +220,1577 @@ test("combobox has correct accessible name when aria-labelledby prop is provided
...
+    );
+    const icon = within(screen.getByRole("presentation")).getByTestId("icon");
+    await user.click(icon);
+
+    expect(onClickFn).toHaveBeenCalled();
+  });
+});
+
+describe("the placement of the list element", () => {
+  it("should set the expected attribute on the list when `listPlacement` is is not passed", async () => {
```
Comment (paulrobinson88) [inline src/components/select/filterable-select/filterable-select.test.tsx:880]: Slight typo
```suggestion
  it("should set the expected attribute on the list when `listPlacement` is not passed", async () => {
```

**Review by damienrobson-sage — APPROVED** [review]

**Review by nuria1110 — APPROVED** [review]

**Review by paulrobinson88 — APPROVED** [review]

**Pull request merged**